tests/cephfs/snapshot_clone/clone_status.py

```diff
@@ -0,0 +1,181 @@
+        config = kw.get("config")
+        clients = ceph_cluster.get_ceph_objects("client")
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        log.info("checking Pre-requisites")
+        if len(clients) < 1:
+            log.info(
+                f"This test requires minimum 1 client nodes.This has only {len(clients)} clients"
+            )
+            return 1
+        default_fs = "cephfs"
+        mounting_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(10))
+        )
+        client1 = clients[0]
+        fs_details = fs_util.get_fs_info(client1)
+        if not fs_details:
+            fs_util.create_fs(client1, "cephfs")
+        subvolumegroup_list = [
+            {"vol_name": default_fs, "group_name": "subvolgroup_clone_status_1"},
+            {"vol_name": default_fs, "group_name": "subvolgroup_clone_status_2"},
+        ]
+        for subvolumegroup in subvolumegroup_list:
+            fs_util.create_subvolumegroup(client1, **subvolumegroup)
+        subvolume = {
+            "vol_name": default_fs,
+            "subvol_name": "subvol_clone_status",
+            "group_name": "subvolgroup_clone_status_1",
+            "size": "5368706371",
+        }
+        fs_util.create_subvolume(client1, **subvolume)
+        log.info("Get the path of sub volume")
+        subvol_path, rc = client1.exec_command(
+            sudo=True,
+            cmd=f"ceph fs subvolume getpath {default_fs} subvol_clone_status subvolgroup_clone_status_1",
+        )
+        kernel_mounting_dir_1 = f"/mnt/cephfs_kernel{mounting_dir}_1/"
+        mon_node_ips = fs_util.get_mon_node_ips()
+        fs_util.kernel_mount(
+            [clients[0]],
+            kernel_mounting_dir_1,
+            ",".join(mon_node_ips),
+            sub_dir=f"{subvol_path.read().decode().strip()}",
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"python3 /home/cephuser/smallfile/smallfile_cli.py --operation create --threads 10 --file-size 400 "
+            f"--files 100 --files-per-dir 10 --dirs-per-dir 2 --top "
+            f"{kernel_mounting_dir_1}",
+            long_running=True,
+        )
+        snapshot = {
+            "vol_name": default_fs,
+            "subvol_name": "subvol_clone_status",
+            "snap_name": "snap_1",
+            "group_name": "subvolgroup_clone_status_1",
+        }
+        fs_util.create_snapshot(client1, **snapshot)
+        clone_status_1 = {
+            "vol_name": default_fs,
+            "subvol_name": "subvol_clone_status",
+            "snap_name": "snap_1",
+            "target_subvol_name": "clone_status_1",
+            "group_name": "subvolgroup_clone_status_1",
+        }
+        fs_util.create_clone(client1, **clone_status_1)
+        fs_util.validate_clone_state(client1, clone_status_1)
```

Comment:
Could we add "pending" & "in-progress" states testing here as well

Reply:
We are checking the Transition states as part of clone_states.py
This test case covers mounting on different platforms(fuse,Kernel) and validating the clone creation